velox/functions/sparksql/RegexFunctions.cpp

```diff
@@ -107,63 +107,91 @@ void ensureRegexIsConstantAndCompatible(
 // If position <= 0, throw error.
 // If position > length string, return string.
 template <typename T>
-struct RegexReplaceFunction {
+struct RegexpReplaceFunction {
   VELOX_DEFINE_FUNCTION_TYPES(T);
 
-  void call(
+  bool callNullable(
       out_type<Varchar>& result,
-      const arg_type<Varchar>& stringInput,
-      const arg_type<Varchar>& pattern,
-      const arg_type<Varchar>& replace) {
-    re2::RE2* patternRegex = getCachedRegex(pattern.str());
-    re2::StringPiece replaceStringPiece = toStringPiece(replace);
+      const arg_type<Varchar>* stringInput,
+      const arg_type<Varchar>* pattern,
+      const arg_type<Varchar>* replace) {
+    if (!stringInput) {
+      result.resize(0);
+      return false;
+    } else if (!pattern || !replace) {
+      result.resize(stringInput->size());
+      std::memcpy(result.data(), stringInput->data(), stringInput->size());
+      return true;
+    }
+    re2::RE2* patternRegex = nullptr;
+    patternRegex = getCachedRegex(pattern->str());
 
-    std::string string(stringInput.data(), stringInput.size());
+    re2::StringPiece replaceStringPiece = toStringPiece(*replace);
+
+    std::string string(stringInput->data(), stringInput->size());
     RE2::GlobalReplace(&string, *patternRegex, replaceStringPiece);
 
     if (string.size()) {
       result.resize(string.size());
       std::memcpy(result.data(), string.data(), string.size());
+      return true;
     } else {
       result.resize(0);
+      return false;
     }
   }
 
-  void call(
+  bool callNullable(
       out_type<Varchar>& result,
-      const arg_type<Varchar>& stringInput,
-      const arg_type<Varchar>& pattern,
-      const arg_type<Varchar>& replace,
-      const arg_type<int64_t>& position) {
-    VELOX_USER_CHECK_GE(position, 1, "regex_replace requires a position >= 1");
+      const arg_type<Varchar>* stringInput,
+      const arg_type<Varchar>* pattern,
+      const arg_type<Varchar>* replace,
+      const arg_type<int64_t>* position) {
+    if (!position) {
+      return callNullable(result, stringInput, pattern, replace);
```

Comment:
Is there a lot of copy-paste between two functions: with and w/o position argument? Couldn't function w/o position call the one with position using position value 1 or similar?